### PR TITLE
[Merged by Bors] - Migrate integration tests

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- Moved tests to operator repo
- Adapted to [ADR018](https://docs.stackable.tech/home/contributor/adr/ADR018-product_image_versioning.html)

Currently versions are provided as e.g. `3.1.0-stackable0` instead of the a fully qualified one like 
`3.1.0-scala2.13-opa_authorizer1.4.0-stackable0.5.0` which is not accepted by the product config semver crate.

Removed from integrations tests repo here: https://github.com/stackabletech/integration-tests/pull/248

## Review Checklist

- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by backstreetkiwi — APPROVED** [review]
Test in Jenkins successful: https://ci.stackable.tech/view/02%20Component%20Tests%20(flex)/job/Kafka%20Operator%20Integration%20Tests%20(flex)/46/

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -33,8 +33,3 @@ repos:
     rev: 4.0.1
     hooks:
       - id: flake8
-
```
Comment (sbernauer) [inline .pre-commit-config.yaml:36]: Same as in https://github.com/stackabletech/trino-operator/pull/212: Intentionally removed? (If yes maybe operator-rs would be a great place :))

File: rust/crd/src/lib.rs [hunk 2]
```diff
@@ -172,11 +182,28 @@ impl KafkaCluster {
...
+    }
+
+    /// Returns our semver representation for product config e.g. 2.8.1
+    pub fn product_version(&self) -> Result<&str, Error> {
+        let image_version = self.image_version()?;
+        image_version
+            .split('-')
+            .collect::<Vec<_>>()
+            .first()
+            .cloned()
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:201]: NIT
```suggestion
            .next()
```

Comment (sbernauer) [inline rust/crd/src/lib.rs:201]: NIT, removes collecting whole list and throwing it away later on
```suggestion
            .next()
```

File: rust/operator/src/kafka_controller.rs [hunk 3]
```diff
@@ -458,11 +490,10 @@ fn build_broker_rolegroup_statefulset(
         .with_context(|| RoleGroupNotFoundSnafu {
             rolegroup: rolegroup_ref.clone(),
         })?;
-    let kafka_version = kafka_version(kafka)?;
-    let image = format!(
-        "docker.stackable.tech/stackable/kafka:{}-stackable0",
-        kafka_version
-    );
+    let kafka_version = kafka
```
Comment (sbernauer) [inline rust/operator/src/kafka_controller.rs:493]: Naming it `image_version` or `kafka_image_version` would be great

Comment (sbernauer) [inline rust/operator/src/kafka_controller.rs:493]: Naming it `image_version` or `kafka_image_version` would be great

**Review by sbernauer — APPROVED** [review]
LGTM % comments

File: .pre-commit-config.yaml (hunk 1 above)
Comment (maltesander) [reply .pre-commit-config.yaml:36]: Yeah we remove it from templating for now.

File: rust/crd/src/lib.rs (hunk 2 above)
Comment (maltesander) [reply rust/crd/src/lib.rs]: Thanks, done!

File: rust/operator/src/kafka_controller.rs (hunk 3 above)
Comment (maltesander) [reply rust/operator/src/kafka_controller.rs]: done

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**